### ISSUE TITLE
How to get screen bounds and change cursors in WinUI3 Desktop applications?

### ISSUE THREAD

**Issue by KanniyappanP** [microsoft/microsoft-ui-xaml#4152]
I need to achieve below requirements in WinUI 3 Desktop environment.

1. How to get display screen bounds?

2. How to change windows cursor type in desktop environment?

I already did this in WinUI UWP applications.

**For Screen bounds,**

`var visibleBounds = Windows.UI.ViewManagement.ApplicationView.GetForCurrentView().VisibleBounds;
var scaleFactor = Windows.Graphics.Display.DisplayInformation.GetForCurrentView().RawPixelsPerViewPixel;
Size screenSize = new Size((visibleBounds.Width scaleFactor), (visibleBounds.Height scaleFactor));`

**For Cursor:**

`Window.Current.CoreWindow.PointerCursor = new CoreCursor(CoreCursorType.SizeNorthwestSoutheast, 0);`

Anyone please suggest how to achieve same requirement in WinUI Desktop applications?

**Comment by StephenLPeters** [comment]
@anawishnoff do we have documentation for this somewhere? @JesseCol Do you know how to do this?

**Comment by mdtauk** [comment]
I think the new Window API has guidance for those functions that call "GetForCurrent" and "CoreWindow"

**Comment by JesseCol** [comment]
@marb2000 , do you have a recommendation here?  Thanks!

**Comment by azchohfi** [comment]
For the Cursor, there is a new API for this:
```csharp
Microsoft.UI.Xaml.UIElement
{
    protected CoreCursor ProtectedCursor { get; set; }
}
```

You can set this property and it will handle it all for you. The only problem is that you need to have a subclass of UIElement to access it.


For DisplayInformation.GetForCurrentView().RawPixelsPerViewPixel, you need to get the XamlRoot.RasterizationScale.

**Comment by KarthikRajaAKR** [comment]
Hi Team,
I've used  `Windows.UI.Core.CoreWindow.GetForCurrentThread().PointerCursor` to get the pointer, but in WinUI3 Preview 4 CoreWindow has been disabled. Is there any alternative available to get the pointer?

**Comment by jschwizer99** [comment]
@azchohfi many thanks for the valuable hint! This replaces 'Windows.UI.Core.CoreWindow.GetForCurrentThread().PointerCursor' for me. But it would be very nice if `ProtectedCursor` was accessible as a property to get it working in XAML without the need of subclassing each element I want to change the cursor.

**Comment by marb2000** [comment]
@jschwizer99 @KanniyappanP and @KarthikRajaAKR it seems that your question was resolved. I'm closing this issue. Feel free to reopen it in case we miss something related or open a new one.

Thanks!

**Comment by KanniyappanP** [comment]
Hi @marb2000 

I have used ProtectedCursor from UIElement as you suggested instead of  using from CoreWindow (CoreWindow.GetForCurrentThread().PointerCursor).

I have used like below. but not working for me. 

` ProtectedCursor = new CoreCursor(CoreCursorType.SizeNorthwestSoutheast, 0);`

Thanks.,

**Comment by bridgesquared** [comment]
Hi @KanniyappanP 

I had the same experience of using `ProtectedCursor` and finding that the mouse pointer didn't change, really wound me up for a while but I needed to give the user the visual feedback of being busy so perservered.

Firstly I can confirm that `ProtectedCursor` does work, so stick with it.

The issue I found was that I had a `UserControl` containing the `Button` that triggered the action that tried to set the cursor. The conflict was that I was setting the UserControl `IsEnabled = false` and that seemed to create a conflict such that the mouse pointer wasn't changing.

I would recommend you go back to small steps, just set the cursor with no other side affects, convince yourself that it works and then build up the functionality you want and find out where the conflict occurs.

Another thought is that you might find that your action is fully synchronous and never yields the thread to change the pointer. You might try inserting a `await Task.Delay(1000)` to force a yield?

Hope that helps, best of luck ...
